src/pokemon.c

```diff
@@ -6424,3 +6424,95 @@ u16 GetSpeciesPreEvolution(u16 species)
 
     return SPECIES_NONE;
 }
+
+u32 CreateCustomMon(u16 species, u8 level, u16 item, u8 ball, u8 nature, u8 abilityNum, u8 gender, u8 *evs, u8 *ivs, u16 *moves, bool8 isShiny)
+{
+    u16 nationalDexNum;
+    int sentToPc;
+    struct Pokemon mon;
+    u32 i;
+
+    // check whether to use a specific nature or a random one
+    if (nature >= NUM_NATURES)
+        nature = Random() % NUM_NATURES;
+
+    // create a Pokémon with basic data
+    if (gender == MON_MALE || gender == MON_FEMALE)
+        CreateMonWithGenderNatureLetter(&mon, species, level, 32, gender, nature, 0);
+    else
+        CreateMonWithNature(&mon, species, level, 32, nature);
+
+    if (isShiny)
+        SetMonData(&mon, MON_DATA_IS_SHINY, &isShiny);
+
+    for (i = 0; i < NUM_STATS; i++)
+    {
+        // EV
+        if (evs[i] <= MAX_PER_STAT_EVS)
+            SetMonData(&mon, MON_DATA_HP_EV + i, &evs[i]);
```

Comment:
Mons can have Illegal EVs. Not sure how much we should care because if a hacker decided to give Illegal EVs they might have good reasons for it

Reply:
I agree with the latter, but I don't have much of an opinion either way. I don't particularly care.

Giving a Pokémon max EV in all 6 stats might be good for certain debugging or research related tasks for whatever that's worth, like seeing how high a Pokémon's stats can get with max EV without having to compile a new build multiple times raising just a few stats at a time and such I guess.

Reply:
Then let's keep it that way.